src/combinatorics/simple_graph/subgraph.lean

```diff
@@ -493,7 +525,144 @@ begin
+    { rw h,
+      simp, },
+    { simp only [Prop.bot_eq_false, singleton_subgraph_adj, pi.bot_apply, iff_false],
+      intro ha,
+      have ha1 := ha.fst_mem,
+      have ha2 := ha.snd_mem,
+      rw [h, set.mem_singleton_iff] at ha1 ha2,
+      subst_vars,
+      exact ha.ne rfl } },
+end
+
+instance nonempty_subgraph_of_adj_verts {v w : V} (hvw : G.adj v w) :
+  nonempty (G.subgraph_of_adj hvw).verts := ⟨⟨v, by simp⟩⟩
+
+@[simp] lemma edge_set_subgraph_of_adj {v w : V} (hvw : G.adj v w) :
+  (G.subgraph_of_adj hvw).edge_set = {⟦(v, w)⟧} :=
+begin
+  ext e,
+  refine e.ind _,
+  simp only [set.mem_singleton_iff, subgraph.mem_edge_set, subgraph_of_adj_adj, eq_comm,
+    iff_self, forall_2_true_iff],
+end
+
+lemma subgraph_of_adj_symm {v w : V} (hvw : G.adj v w) :
+  G.subgraph_of_adj hvw.symm = G.subgraph_of_adj hvw :=
+by ext; simp [or_comm, and_comm]
+
+@[simp] lemma map_subgraph_of_adj (f : G →g G')
+  {v w : V} (hvw : G.adj v w) :
+  subgraph.map f (G.subgraph_of_adj hvw) = G'.subgraph_of_adj (f.map_adj hvw) :=
+begin
+  ext,
+  { simp only [subgraph.map_verts, subgraph_of_adj_verts, set.mem_image,
+      set.mem_insert_iff, set.mem_singleton_iff],
+    split,
+    { rintro ⟨u, rfl|rfl, rfl⟩; simp },
+    { rintro (rfl|rfl),
+      { use v, simp },
+      { use w, simp } } },
+  { simp only [relation.map, subgraph.map_adj, subgraph_of_adj_adj, quotient.eq, sym2.rel_iff],
+    split,
+    { rintro ⟨a, b, (⟨rfl,rfl⟩|⟨rfl,rfl⟩), rfl, rfl⟩; simp },
+    { rintro (⟨rfl,rfl⟩|⟨rfl,rfl⟩),
+      { use [v, w], simp },
+      { use [w, v], simp } } }
+end
+
+lemma neighbor_set_subgraph_of_adj_subset {u v w : V} (hvw : G.adj v w) :
+  (G.subgraph_of_adj hvw).neighbor_set u ⊆ {v, w} :=
+(G.subgraph_of_adj hvw).neighbor_set_subset_verts _
+
+@[simp] lemma neighbor_set_fst_subgraph_of_adj {v w : V} (hvw : G.adj v w) :
+  (G.subgraph_of_adj hvw).neighbor_set v = {w} :=
+begin
+  ext u,
+  suffices : w = u ↔ u = w, by simpa [hvw.ne.symm] using this,
+  rw eq_comm,
+end
+
+@[simp] lemma neighbor_set_snd_subgraph_of_adj {v w : V} (hvw : G.adj v w) :
+  (G.subgraph_of_adj hvw).neighbor_set w = {v} :=
+begin
+  rw subgraph_of_adj_symm hvw.symm,
+  exact neighbor_set_fst_subgraph_of_adj hvw.symm,
+end
+
+@[simp] lemma neighbor_set_subgraph_of_adj_of_ne_of_ne {u v w : V} (hvw : G.adj v w)
+  (hv : u ≠ v) (hw : u ≠ w) :
+  (G.subgraph_of_adj hvw).neighbor_set u = ∅ :=
+by { ext, simp [hv.symm, hw.symm] }
+
+lemma neighbor_set_subgraph_of_adj [decidable_eq V] {u v w : V} (hvw : G.adj v w) :
+  (G.subgraph_of_adj hvw).neighbor_set u =
+  (if u = v then {w} else ∅) ∪ (if u = w then {v} else ∅) :=
+by split_ifs; subst_vars; simp [*]
+
+lemma singleton_subgraph_fst_le_subgraph_of_adj {u v : V} {h : G.adj u v} :
+  G.singleton_subgraph u ≤ G.subgraph_of_adj h :=
+by split; simp [-set.bot_eq_empty]
+
+lemma singleton_subgraph_snd_le_subgraph_of_adj {u v : V} {h : G.adj u v} :
+  G.singleton_subgraph v ≤ G.subgraph_of_adj h :=
+by split; simp [-set.bot_eq_empty]
+
```

Comment:
I suppose there should also be the result that `G.subgraph_of_adj h = (\top : G.subgraph).induce {u, v}`.